catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/FutureDateValidator.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation.impl.validator;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Optional;
+
+import com.google.common.base.Preconditions;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.validation.AttributeValidator;
+import ddf.catalog.validation.impl.report.AttributeValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.AttributeValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation.Severity;
+
+public class FutureDateValidator implements AttributeValidator {
```

Comment:
The logic for this and the `FutureDateValidator` are nearly identical. I'd like to see an `AbstractDateValidator` that provides the boilerplate of the functionality where two params are passed in: a lambda for the actual test function and the portion of the message that is different. This would help if we someday wanted to add a `BetweenDateValidator` as well.